### PR TITLE
Upgrade to the new IPLD spec

### PR DESCRIPTION
**Pull request by dignifiedquire**
- [x] Setup dignified.js
- [x] Upgrade all dependencies
- [x] Bring `ipld.marshal` and `ipld.unmarshal` up to date with the new spec
- [x] Add `ipld.multihash` to generate the canoncial multihash
- [x] Use Cbor tag `258` as specified in https://github.com/mildred/specs/blob/cbor-tag-merkle-link/merkledag/cbor_tag_merkle_link.md

### Items below will not be implemented in this repo

~~Add `ipld.store` to store objects either in memory or in ipfs to be used with the resolver~~
~~Add `ipld.resolve` to resolve paths (and links of course)~~
~~Add cli tool for interacting~~

### CODE REVIEW

**Comment by dignifiedquire** [comment]
@diasdavid please enable travis

**Comment by diasdavid** [comment]
> Add ipld.store to store objects either in memory or in ipfs to be used with the resolver
> Add ipld.resolve to resolve paths

Note that require('ipld') should be just the data struct and its features, when we can have the IPLD service (similar to DAG and DAGService) that plugs it into a IPFS Repo for storing and/or bitswap for resolving. We don't want to enforce strictly that storing and resolving should be through IPFS.

the IPLD-Service can be behind require('ipld').Service or even as a separate module :)

**Comment by diasdavid** [comment]
Travis enabled :)

**Comment by diasdavid** [comment]
moved this repo to IPFS org

//cc @jbenet @nicola

**Comment by nicola** [comment]
This work looks really great, I will have to go through it in more detail.
I have wrote last night a trivial demo of the scheme that I proposed in https://github.com/ipfs/specs/issues/90 built on json (for the sake of showing how I would like ipld to be)

Last chat I had with Juan there was the issue of `//` and `@link`, hence why I started working on my proposal which avoids them (which I think, looking at the implementation and other comments, that we both converged to the same scheme - which is awesome)

I hope to meet in person very soon and describe the full vision and align with you (and participate on this). This is great and it will be a key piece of schemes using merkle dags/ipfs

File: test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+describe('IPLD', () => {
+  it('constructor', () => {
+    var file = new IPLD({
+      name: 'hello.txt',
+      size: 11
+    })
+
+    expect(
+      file.unmarshal()
```
Comment (diasdavid) [inline test/index.spec.js]: Was it marsheled though?

unmarshal should be something like:

`var file = IPLD.unmarshal(cborSerializedIPLDObj)`

File: test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  it('hash', () => {
+    var file = new IPLD({
+      name: 'hello.txt',
+      size: 11
+    })
+
+    expect(
+      file.hash
+    ).to.be.eql(
+      'QmQtX5JVbRa25LmQ1LHFChkXWW5GaWrp7JpymN4oPuBSmL'
```
Comment (diasdavid) [inline test/index.spec.js]: 👍

File: test/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  })
+
+  it('hash', () => {
+    var file = new IPLD({
+      name: 'hello.txt',
+      size: 11
+    })
+
+    expect(
+      file.hash
```
Comment (diasdavid) [inline test/index.spec.js]: let's check if this can also be a function `file.hash()`

**Comment by dignifiedquire** [comment]
@diasdavid updated, I think this is ready for the first merge as discussed.

@nicola thanks for the kind words, I will be looking into path resolving in the next couple of days.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  - '5'
+  - stable
+
+addons:
+  firefox: 'latest'
+
+before_script:
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
+
```
Comment (diasdavid) [inline .travis.yml]: missing the `npm i npm -g`, making tests fail on `Node.js 4`

**Comment by diasdavid** [comment]
released version 0.5.0 :)

**Comment by dignifiedquire** [comment]
:tada:

**Pull request merged**